### DISCUSSION TITLE
Built in EPUB reader questions

### DISCUSSION THREAD

**Discussion by angryviking** [advplyr/audiobookshelf#2029] [General]
I have just started getting into audiobooks and audiobookshelf seems to work well for those, so thank you for creating this project.
I was also excited to learn that audiobookshelf supports my EPUB files and I would like to use it replace calibre-web for a local web based ebook reader/library.

Currently, the EPUB reader in calibre-web seems to work better than the one included with audiobookshelf. Here a few screenshots for examples. Both audiobookshelf and calibre-web are using the same .epub files and running on the same docker platform.

Audiobookshelf -
![image](https://github.com/advplyr/audiobookshelf/assets/28925820/0bf3ecc5-1ef5-49a8-8b4d-e987503c2dff)
![image](https://github.com/advplyr/audiobookshelf/assets/28925820/07909c4c-ac9e-49dd-9b3a-87b04e75bc88)

Calibre-web -
![image](https://github.com/advplyr/audiobookshelf/assets/28925820/16bc73ba-64a3-4f6f-804a-20e2fea0d572)
![image](https://github.com/advplyr/audiobookshelf/assets/28925820/902dad53-3d81-4153-9fd2-bd55f665fdcc)

It looks like the audiobookshelf reader squishes inline images. I have had other EPUB files not open at all in audiobookshelf, but the same file opens in calibre-web fine.
I was wondering if the reader that comes with calibre-web could be included with audiobookshelf. I am assuming that epub reader is not developed by calibre-web, but a third-party epub reader. I could be completely wrong about that as I haven't looked it at all.
It is also possible that there is some configuration option that I haven't discovered that needs to be set or tweaked to get the books to display better.

Thank you

**Comment by nichwall** [comment]
Does the Calibre-web reader track your progress? I didn't think it did but I haven't used Calibre-web in about a year.

One of the difficult parts was getting an epub reader that worked on all of the clients (web and native mobile apps), tracked progress, and was relatively simple/lightweight. There are definitely issues with the 3rd party epubjs library currently being used, but a better one hasn't been found yet.

**Comment by angryviking** [comment]
I wasn't sure about the calibre-web reader keeping track of progress, so I checked it out. Initially, I thought it did because I logged out of calibre-web, closed the tab and reopened it. When I logged back in and opened the book, it opened to same place I had left it.

Unfortunately, that may have just been the browser cache. When I logged into calibre-web with a different browser, my progress was not kept. So you are probably correct about the progress.

**Reply by nichwall** [reply]
Ah, darn. Thanks for checking! Their reader is definitely something to look into, just not sure exactly when that'll happen. ABS just barely added tracking reading reading progress for ebooks a couple of months ago and there's currently bigger issues being worked on at the moment. Hopefully that can get revised soon. :)
